icefoganalytics/travel-authorization
# Set Up Back-End Linting

Fixes https://github.com/icefoganalytics/travel-authorization/issues/184

Relates to:
- https://github.com/icefoganalytics/internal-data-portal/commit/159c2b4f0eaa6cdd6fb55bf368d2e3225b6917ee
- https://github.com/icefoganalytics/internal-data-portal/pull/31

# Context

Set up linting in the back-end. 
This vastly improves code quality, and makes it easier to find bugs.

# Implementation

⚙️ Set up minimal eslint config for back-end api.

# Screenshots

`dev api npm run lint`
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/ac9fd8ef-9a41-47f0-8127-86631cafdde2)

A file with some linting errors now showing up in VS Code.
![image](https://github.com/icefoganalytics/travel-authorization/assets/23045206/28f9226d-2364-48b3-a487-8c309d97803f)


# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. From the bash console run `dev api npm run lint`
5. Go to `api/src/services/per-diems-service.ts` in VS Code and check that it has a bunch of warnings.
